### PR TITLE
kernel: bump 5.10 to 5.10.42

### PR DESCRIPTION
**Pull request by rsalvaterra**
Deleted (reverse-appliable):
generic/pending-5.10/110-perf-jevents-fix-getting-maximum-number-of-fds.patch

Manually refreshed:
ramips/patches-5.10/835-asoc-add-mt7620-support.patch

Automatically refreshed:
ath79/patches-5.10/921-serial-core-add-support-for-boot-console-with-arbitr.patch
generic/backport-5.10/343-netfilter-nft_flow_offload-handle-netdevice-events-f.patch
generic/backport-5.10/600-v5.12-net-extract-napi-poll-functionality-to-\_\_napi_poll.patch
generic/backport-5.10/601-v5.12-net-implement-threaded-able-napi-poll-loop-support.patch
generic/backport-5.10/602-v5.12-net-add-sysfs-attribute-to-control-napi-threaded-mod.patch
generic/backport-5.10/603-v5.12-net-fix-race-between-napi-kthread-mode-and-busy-poll.patch
generic/backport-5.10/604-v5.12-net-fix-hangup-on-napi_disable-for-threaded-napi.patch
generic/backport-5.10/610-v5.13-01-netfilter-flowtable-separate-replace-destroy-and-sta.patch
generic/backport-5.10/610-v5.13-04-netfilter-flowtable-consolidate-skb_try_make_writabl.patch
generic/backport-5.10/610-v5.13-06-netfilter-flowtable-move-FLOW_OFFLOAD_DIR_MAX-away-f.patch
generic/backport-5.10/610-v5.13-07-netfilter-flowtable-fast-NAT-functions-never-fail.patch
generic/backport-5.10/610-v5.13-16-net-dsa-resolve-forwarding-path-for-dsa-slave-ports.patch
generic/backport-5.10/610-v5.13-17-netfilter-flowtable-add-xmit-path-types.patch
generic/backport-5.10/610-v5.13-18-netfilter-flowtable-use-dev_fill_forward_path-to-obt.patch
generic/backport-5.10/610-v5.13-19-netfilter-flowtable-use-dev_fill_forward_path-to-obt.patch
generic/backport-5.10/610-v5.13-20-netfilter-flowtable-add-vlan-support.patch
generic/backport-5.10/610-v5.13-26-netfilter-nft_flow_offload-use-direct-xmit-if-hardwa.patch
generic/backport-5.10/610-v5.13-27-netfilter-flowtable-bridge-vlan-hardware-offload-and.patch
generic/backport-5.10/610-v5.13-30-dsa-slave-add-support-for-TC_SETUP_FT.patch
generic/backport-5.10/610-v5.13-31-net-ethernet-mtk_eth_soc-fix-parsing-packets-in-GDM.patch
generic/backport-5.10/610-v5.13-32-net-ethernet-mtk_eth_soc-add-support-for-initializin.patch
generic/backport-5.10/610-v5.13-33-net-ethernet-mtk_eth_soc-add-flow-offloading-support.patch
generic/backport-5.10/610-v5.13-38-net-ethernet-mtk_eth_soc-unmap-RX-data-before-callin.patch
generic/backport-5.10/610-v5.13-39-net-ethernet-mtk_eth_soc-fix-build_skb-cleanup.patch
generic/backport-5.10/610-v5.13-40-net-ethernet-mtk_eth_soc-use-napi_consume_skb.patch
generic/backport-5.10/610-v5.13-42-net-ethernet-mtk_eth_soc-remove-unnecessary-TX-queue.patch
generic/backport-5.10/610-v5.13-43-net-ethernet-mtk_eth_soc-use-larger-burst-size-for-Q.patch
generic/backport-5.10/610-v5.13-45-net-ethernet-mtk_eth_soc-implement-dynamic-interrupt.patch
generic/backport-5.10/610-v5.13-46-net-ethernet-mtk_eth_soc-cache-HW-pointer-of-last-fr.patch
generic/backport-5.10/610-v5.13-47-net-ethernet-mtk_eth_soc-only-read-the-full-RX-descr.patch
generic/backport-5.10/610-v5.13-48-net-ethernet-mtk_eth_soc-reduce-unnecessary-interrup.patch
generic/backport-5.10/610-v5.13-49-net-ethernet-mtk_eth_soc-rework-NAPI-callbacks.patch
generic/backport-5.10/610-v5.13-50-net-ethernet-mtk_eth_soc-set-PPE-flow-hash-as-skb-ha.patch
generic/backport-5.10/610-v5.13-51-net-ethernet-mtk_eth_soc-use-iopoll.h-macro-for-DMA-.patch
generic/backport-5.10/771-v5.12-net-dsa-be-louder-when-a-non-legacy-FDB-operation-fa.patch
generic/backport-5.10/772-v5.12-net-dsa-don-t-use-switchdev_notifier_fdb_info-in-dsa.patch
generic/backport-5.10/773-v5.12-net-dsa-move-switchdev-event-implementation-under-th.patch
generic/backport-5.10/774-v5.12-net-dsa-exit-early-in-dsa_slave_switchdev_event-if-w.patch
generic/backport-5.10/775-v5.12-net-dsa-listen-for-SWITCHDEV_-FDB-DEL-_ADD_TO_DEVICE.patch
generic/backport-5.10/781-v5.13-net-dsa-mt7530-Add-support-for-EEE-features.patch
generic/hack-5.10/650-netfilter-add-xt_FLOWOFFLOAD-target.patch
generic/hack-5.10/661-use_fq_codel_by_default.patch
generic/pending-5.10/630-packet_socket_type.patch
generic/pending-5.10/680-NET-skip-GRO-for-foreign-MAC-addresses.patch
generic/pending-5.10/765-net-dsa-Include-local-addresses-in-assisted-CPU-port.patch
generic/pending-5.10/766-net-dsa-Include-bridge-addresses-in-assisted-CPU-por.patch
generic/pending-5.10/767-net-dsa-Sync-static-FDB-entries-on-foreign-interface.patch
mediatek/patches-5.10/410-bt-mtk-serial-fix.patch
mediatek/patches-5.10/700-net-ethernet-mtk_eth_soc-add-support-for-coherent-DM.patch
ramips/patches-5.10/323-mt7621-memory-detect.patch
ramips/patches-5.10/700-net-ethernet-mediatek-support-net-labels.patch

Run-tested:
ramips/mt7621 (Redmi AC2100)

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

### CODE REVIEW

File: target/linux/ramips/patches-5.10/835-asoc-add-mt7620-support.patch [hunk 1]
```diff
@@ -20,22 +20,14 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
 
 --- a/arch/mips/ralink/of.c
 +++ b/arch/mips/ralink/of.c
-@@ -13,6 +13,7 @@
+@@ -14,6 +14,7 @@
  #include <linux/of_fdt.h>
  #include <linux/kernel.h>
  #include <linux/memblock.h>
 +#include <linux/module.h>
```
Comment (hauke) [inline target/linux/ramips/patches-5.10/835-asoc-add-mt7620-support.patch:27]: I think this include is not needed any more, upstream now included ```#include <linux/export.h>```
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/arch/mips/ralink/of.c?id=v5.10.42&id2=v5.10.41

Comment (rsalvaterra) [reply target/linux/ramips/patches-5.10/835-asoc-add-mt7620-support.patch:27]: Indeed, I missed that. Will fix and update, thanks!

Comment (rsalvaterra) [reply target/linux/ramips/patches-5.10/835-asoc-add-mt7620-support.patch]: Fixed, build-tested and updated.

**Comment by adschm** [comment]
Two general remarks:

> Deleted (reverse-appliable):

Do you actually check the content there? It's not frequent, but occasionally "reverse-applicable" patches still need manual care, and cannot just be removed without introducing problems.

> Automatically refreshed:

Personally, I'd completely remove that, it just bloats the commit message. The patches can be found from 'git diff' ...

**Comment by rsalvaterra** [comment]
Hi, Adrian,

A sexta, 4/06/2021, 18:37, Adrian Schmutzler ***@***.***>
escreveu:

> Two general remarks:
>
> Deleted (reverse-appliable):
>
> Do you actually check the content there? It's not frequent, but
> occasionally "reverse-applicable" patches still need manual care, and
> cannot just be removed without introducing problems.
>
Yes, I do:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.10.y&id=e31975c346bf653276423a5d1f25f1c597c522ba

Automatically refreshed:
>
> Personally, I'd completely remove that, it just bloats the commit message.
> The patches can be found from 'git diff' ...
>
Agreed, will do.

Thanks,
Rui

**Comment by adschm** [comment]
Thanks, perfect. :-)

**Comment by rsalvaterra** [comment]
Ok, done, guys!

**Comment by hauke** [comment]
Thank you for the patch, I applied it to master.

**Comment by abajk** [comment]
Hi,
I see a large performance drop between 5.10.41 and 5.10.42 on Lantiqu from DSA.
The results of NAT benchmark (with flow offloading):
 * 5.4.124    515 Mb/s
 * 5.10.41    570 Mb/s
 * 5.10.42    250 Mb/s

All settings are the same. Does anyone see this drop in performance on other targets?

**Comment by Vladdrako** [comment]
@abajk Not sure about speed, but on Xiaomi Mi Router 3 Pro I have a high ping issue. Reverting to 5.10.41 fixed it. So this update is buggy.

P.S. Offtop, but after deleting the MESH network I see:
<pre>
[   99.235508] do_page_fault(): sending SIGSEGV to wpa_supplicant for invalid read access from 0000006c
[   99.244708] epc = 77aac758 in libubus.so.20210603[77aaa000+15000]
[   99.251151] ra  = 77aac754 in libubus.so.20210603[77aaa000+15000]
</pre>

**Comment by rsalvaterra** [comment]
This is very odd. @Vladdrako, I have a Redmi AC2100 (same SoC as the R3P) and I'm not seeing any issues at all. Granted, I don't use flow offloading (as it interferes with SQM) or mesh.
```
root@rocket:~# uname -a
Linux rocket.lan 5.10.42 #0 SMP Tue Jun 8 13:42:44 2021 mips GNU/Linux
root@rocket:~# ping -c 8 9.9.9.9
PING 9.9.9.9 (9.9.9.9): 56 data bytes
64 bytes from 9.9.9.9: seq=0 ttl=60 time=8.261 ms
64 bytes from 9.9.9.9: seq=1 ttl=60 time=8.770 ms
64 bytes from 9.9.9.9: seq=2 ttl=60 time=7.540 ms
64 bytes from 9.9.9.9: seq=3 ttl=60 time=8.333 ms
64 bytes from 9.9.9.9: seq=4 ttl=60 time=8.689 ms
64 bytes from 9.9.9.9: seq=5 ttl=60 time=7.692 ms
64 bytes from 9.9.9.9: seq=6 ttl=60 time=8.094 ms
64 bytes from 9.9.9.9: seq=7 ttl=60 time=7.633 ms

--- 9.9.9.9 ping statistics ---
8 packets transmitted, 8 packets received, 0% packet loss
round-trip min/avg/max = 7.540/8.126/8.770 ms
root@rocket:~# 
```

**Comment by Vladdrako** [comment]
@rsalvaterra Tested 5.10.42 again and yes, HW NAT is broken. Again xD No ping issue without HW/SW NAT.

**Comment by abajk** [comment]
@Vladdrako
I have bisected upstram commit [1]. Can you confirm that reversing this commit also fixes your high ping issue?

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-5.10.y&id=6d6bc8c75290866e59ee25ab6bb0114eb166b980

**Comment by Vladdrako** [comment]
> I have bisected upstram commit [1]. Can you confirm that reverting this commit also fixes your high ping issue?

Yes, now HW NAT works fine, thank you :)

**Comment by rsalvaterra** [comment]
Reverting this commit, however, is [not the correct solution](https://marc.info/?l=linux-netdev&m=162370772126322&w=4). It's the `xt_flowoffload` module, which is actually an OpenWrt hack (at least according to [its location in the source tree](https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=target/linux/generic/hack-5.10/650-netfilter-add-xt_FLOWOFFLOAD-target.patch;h=44e897f10a12cc57d006b76c5b03350921231960;hb=HEAD)) that needs to be fixed. @nbd168?

**Comment by Vladdrako** [comment]
> Reverting this commit, however, is [not the correct solution](https://marc.info/?l=linux-netdev&m=162370772126322&w=4). It's the `xt_flowoffload` module, which is actually an OpenWrt hack (at least according to [its location in the source tree](https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=target/linux/generic/hack-5.10/650-netfilter-add-xt_FLOWOFFLOAD-target.patch;h=44e897f10a12cc57d006b76c5b03350921231960;hb=HEAD)) that needs to be fixed.
Yeah, I know, but leaving correct fix for smart guy's ;)

**Pull request closed without merging**